### ISSUE TITLE
Value gets reset when using attr-for-value different than "value"

### ISSUE THREAD

**Issue by ronnyroeller** [vaadin/vaadin-combo-box#778]
### Description
Using a custom input element with a different attribute for value than `value` leads to the input being reset to `undefined` whenever the user starts typing.

Renaming the property to `value` makes it work (but that's of course difficult if one can't control the input element).

This seemed to be caused by the `ComboBoxMixin`, which hardcodes the attribute name to `value`: https://github.com/vaadin/vaadin-combo-box/blob/master/src/vaadin-combo-box-mixin.html#L607
The line sets the value of the combo box always to `undefined` (because the `value` property doesn't exist).

### Expected outcome
Value from the input box is not overwritten and used by the combo-box.

### Actual outcome
Value is reset to `undefined` whenever the user starts typing.

### Live Demo
https://glitch.com/edit/#!/peaceful-consonant?path=index.html:25:0

### Steps to reproduce
Create a custom input element that exposes its value as a property that is called different than `value`:
```javascript
import {PolymerElement, html} from '@polymer/polymer/polymer-element';
import '@polymer/iron-input';

class MyInput extends PolymerElement {
	static get template() {
		return html`
		<iron-input bind-value="{{search}}">
			<input>
		</iron-input>`;
	}

	static get properties() {
		return {
			search: {
				notify: true,
				type: String,
			},
		};
	}
}
window.customElements.define('my-input', MyInput);
```

Use this custom input element with the Vaadin combo-box:
```html
<dom-bind id="vaadin">
	<template>
		<vaadin-combo-box-light attr-for-value="search" items="[[items]]">
			<my-input class="input"></my-input>
		</vaadin-combo-box-light>
	</template>
</dom-bind>

<script>
	var scope = document.querySelector('#vaadin');
	scope.items = ['apples', 'pears', 'bananas'];
</script>
```

### Browsers Affected
- [x] Chrome
- [x] Firefox
- [x] Safari
- [x] Edge
- [x] IE 11
- [x] iOS Safari
- [x] Android Chrome

**Comment by web-padawan** [comment]
Nice catch. We should also improve the tests for `vaadin-combo-box-light`.

**Comment by Haprog** [comment]
@ronnyroeller Thanks for the detailed issue description and steps to reproduce. Though I couldn't get the live demo in glitch.com working, but I was able to reproduce it locally.

The above PR contains a test and a fix for this. It's just waiting for review.

**Comment by Haprog** [comment]
This has been fixed in [v4.2.8](https://github.com/vaadin/vaadin-combo-box/releases/tag/v4.2.8).
